cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionUtils.java

```diff
@@ -353,36 +353,68 @@ public static void processAlterPartitionReassignmentsResult(AlterPartitionReassi
+                                                   noReassignmentToCancel,
+                                                   transformToTopicPartitionByKafkaFuture(result.values()));
       }
     }
   }
 
+  private static Map<KafkaFuture<Void>, Set<TopicPartition>> transformToTopicPartitionByKafkaFuture(
+      Map<TopicPartition, KafkaFuture<Void>> kafkaFutureByTopicPartition) {
+
+    Map<KafkaFuture<Void>, Set<TopicPartition>> topicPartitionByKafkaFuture = new HashMap<>();
+    kafkaFutureByTopicPartition.forEach((tp, kafkaFuture) -> {
+      topicPartitionByKafkaFuture.computeIfAbsent(kafkaFuture, kf -> new HashSet<>()).add(tp);
+    });
+    return topicPartitionByKafkaFuture;
+  }
+
+  private static void handleAlterPartitionReassignmentsException(Exception exception,
+                                                                 KafkaFuture<Void> finishedKafkaFuture,
+                                                                 Set<TopicPartition> deleted,
+                                                                 Set<TopicPartition> dead,
+                                                                 Set<TopicPartition> noReassignmentToCancel,
+                                                                 Map<KafkaFuture<Void>, Set<TopicPartition>> topicPartitionByKafkaFuture) {
+
+    if (exception instanceof TimeoutException || exception instanceof InterruptedException) {
+      // May indicate transient (e.g. network) issues and might require a task re-execution -- i.e. handled in Executor.
+      // If this is observed frequently, we may need to bump up EXECUTION_TASK_FUTURE_ERROR_VERIFICATION_TIMEOUT_MS.
+      LOG.warn("Failed to process AlterPartitionReassignmentsResult.", exception);
+      return;
+    }
+    if (!(exception instanceof ExecutionException)) {
+      return;
+    }
+    ExecutionException ee = (ExecutionException) exception;
+    if (Errors.INVALID_REPLICA_ASSIGNMENT.exception().getClass() == ee.getCause().getClass()) {
+      dead.addAll(topicPartitionByKafkaFuture.get(finishedKafkaFuture));
+      LOG.debug("Replica reassignment failed due to dead destination broker(s).", ee);
+    } else if (Errors.UNKNOWN_TOPIC_OR_PARTITION.exception().getClass() == ee.getCause().getClass()) {
+      deleted.addAll(topicPartitionByKafkaFuture.get(finishedKafkaFuture));
+      LOG.debug("Replica reassignment failed due to its topic deletion.", ee);
+    } else if (Errors.NO_REASSIGNMENT_IN_PROGRESS.exception().getClass() == ee.getCause().getClass()) {
+      // Attempt to cancel/rollback a reassignment that does not exist.
+      noReassignmentToCancel.addAll(topicPartitionByKafkaFuture.get(finishedKafkaFuture));
+      LOG.debug("Rollback failed due to lack of corresponding ongoing replica reassignment.", ee);
+    } else {
+      // Not expected to happen.
+      throw new IllegalStateException("Encountered an unknown execution exception.", ee);
+    }
+  }
```

Comment:
This patch drops the name of the affected partitions from the exception logs. Can we add those?